Go1.12 strings.Map breaks package usage of strings.ToUpper and strings.ToLower

Go1.12 CL https://golang.org/cl/131495 will break this package's use of upper() and lower() calls.  This is exhibited in the following test failure ...

```
$ go version
go version go1.12beta1 linux/386
$ go test
--- FAIL: TestToGoName (0.03s)
    util_test.go:66:
                Error Trace:    util_test.go:66
                Error:          Not equal:
                                expected: "X日本語sample2Text"
                                actual  : "X�\xbf\xbd��本語sample2Text"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -X日本語sample2Text
                                +X�����本語sample2Text
                Test:           TestToGoName
    util_test.go:66:
                Error Trace:    util_test.go:66
                Error:          Not equal:
                                expected: "X日本語findThingByID"
                                actual  : "X�\xbf\xbd��本語findThingByID"

                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1 @@
                                -X日本語findThingByID
                                +X�����本語findThingByID
                Test:           TestToGoName
FAIL
exit status 1

```
The issue has to do with this package passing in str[:1] to upper() and lower() assuming that slicing out the first byte is a valid unicode character and if not, the calls to strings.ToUpper or strings.ToLower will return back the same byte and hence can be reused.  The new behavior of strings.ToUpper and strings.ToLower returns back utf8.RuneError if byte is not a valid unicode character instead and breaks the logic in callers when concatenating to another string slice. 

For example -- https://github.com/go-openapi/swag/blob/v0.18.0/util.go#L332-L337.

--- a/util.go
+++ b/util.go
@@ -222,7 +222,7 @@ func lower(str string) string {
 
 // Camelize an uppercased word
 func Camelize(word string) (camelized string) {
-	for pos, ru := range word {
+	for pos, ru := range []rune(word) {
 		if pos > 0 {
 			camelized += string(unicode.ToLower(ru))
 		} else {
@@ -278,7 +278,7 @@ func ToHumanNameTitle(name string) string {
 	for _, w := range in {
 		uw := upper(w)
 		if !isInitialism(uw) {
-			out = append(out, upper(w[:1])+lower(w[1:]))
+			out = append(out, Camelize(w))
 		} else {
 			out = append(out, w)
 		}
@@ -296,7 +296,7 @@ func ToJSONName(name string) string {
 			out = append(out, lower(w))
 			continue
 		}
-		out = append(out, upper(w[:1])+lower(w[1:]))
+		out = append(out, Camelize(w))
 	}
 	return strings.Join(out, "")
 }
@@ -322,19 +322,17 @@ func ToGoName(name string) string {
 		uw := upper(w)
 		mod := int(math.Min(float64(len(uw)), 2))
 		if !isInitialism(uw) && !isInitialism(uw[:len(uw)-mod]) {
-			uw = upper(w[:1]) + lower(w[1:])
+			uw = Camelize(w)
 		}
 		out = append(out, uw)
 	}
 
 	result := strings.Join(out, "")
 	if len(result) > 0 {
-		ud := upper(result[:1])
-		ru := []rune(ud)
-		if unicode.IsUpper(ru[0]) {
-			result = ud + result[1:]
+		if unicode.IsUpper([]rune(result)[0]) {
+			result = result
 		} else {
-			result = "X" + ud + result[1:]
+			result = "X" + result
 		}
 	}
 	return result

--- a/util.go
+++ b/util.go
@@ -329,9 +329,7 @@ func ToGoName(name string) string {
 
 	result := strings.Join(out, "")
 	if len(result) > 0 {
-		if unicode.IsUpper([]rune(result)[0]) {
-			result = result
-		} else {
+		if !unicode.IsUpper([]rune(result)[0]) {
 			result = "X" + result
 		}
 	}

